cpp/ql/test/query-tests/Critical/MemoryFreed/test_free.cpp

```diff
@@ -227,3 +227,21 @@ void test_ms_free(void * memory_descriptor_list) {
     MmFreePagesFromMdl(memory_descriptor_list); //GOOD
     ExFreePool(memory_descriptor_list); // GOOD
 }
+
+void test_loop3(char ** a, char ** b) {
+    if (*a) {
+        free(*a);
+        a++;
+    }
+    use(*a); // GOOD [FALSE POSITIVE]
+
+    for (;*b; b++) {
+        free(*b);
+    }
+    use(*b); // GOOD [FALSE POSITIVE]
+}
+
+void test_deref(char **a) {
+    free(*a);
+    use(*a); // GOOD [FALSE POSITIVE]
```

Comment:
The other two I understand, as you're conceptually moving along to the next element in an array of pointers, and then try to access/free that next element. This is not happening here though. Is the idea of this being an FP just that we wrote `*a` and not `**a`?

Reply:
Yeah, exactly. This is supposed to not have a result:
```cpp
free(a);
printf("I just free'd %p", a);
```
because you're not reading the memory that was deallocated. And indeed, we're not seeing FPs in that case. As this test demonstrates, though, we're seeing FPs when we replace `a` with `*a`